server/src/main/java/org/opensearch/search/lookup/SearchLookup.java

```diff
@@ -76,7 +83,8 @@ public class SearchLookup {
     public SearchLookup(
         MapperService mapperService,
         BiFunction<MappedFieldType, Supplier<SearchLookup>, IndexFieldData<?>> fieldDataLookup,
-        @Nullable String[] types
+        @Nullable String[] types,
+        int shardId
```

Comment:
Default to `UNKNOWN_SHARD_ID` to preserve the interface?

Reply:
I don't think it is a good idea because in this case someone will be able to easily create a `SearchLookup` instance without a shard id. Then it will lead to a runtime error in plugins that depend on a shard id value.

Reply:
Is this class exposed to plugins directly? Does this need to handle compatibility?